lib/Conversion/XTenNNToTorch.cpp

```diff
@@ -37,6 +37,19 @@ using namespace mlir::torch;
 
 namespace {
 
+Type toTorchTensorTypeCast(PatternRewriter &rewriter, Type ty) {
+  auto tensorTy = ty.cast<TensorType>();
+
+  auto elementType = dyn_cast<IntegerType>(tensorTy.getElementType());
+  if (elementType && elementType.isSignlessInteger()) {
+    elementType = rewriter.getIntegerType(tensorTy.getElementType().getIntOrFloatBitWidth(), true);
+    return Torch::ValueTensorType::get(
+        ty.getContext(), tensorTy.getShape(), elementType);
+  }
+  return Torch::ValueTensorType::get(
+      ty.getContext(), tensorTy.getShape(), tensorTy.getElementType());
+}
+
```

Comment:
Maybe make use of your new function inside `Value toTorchTensorTypeCast(PatternRewriter &rewriter, Value input)`, it seems to currently basically duplicate it.